canonical-web-and-design/snapcraft.io
# Release issues

## Done

- Convert `ReleasesController` to functional component
- Added setup into an effect on first load
- Added loading state
- Added `fetchMissingRevision` that's called from `getReleaseDataFromChannelMap`
- Update `getReleaseDataFromChannelMap` to return an optional set of revisions that were missing

This is the minimum amount of work to fix this. There should be a follow up to move this stuff into proper actions and reducers. Unfortunately the way the code is setup now we rely too heavily on the initial state.

## Issue / Card

Fixes #2909 

## QA

- Pull the branch
- Run the site using the dotrun snap with `$ dotrun`
- View the site locally in your web browser at: http://0.0.0.0:8004/
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- Visit http://0.0.0.0:8004/maas/releases
- Change to the 2.7 track
- See that stable has revisions
